### PR TITLE
Fix template priority to be from current vdc

### PR DESCRIPTION
**Pull request by ivanayov**

### CODE REVIEW

**Comment by andreaturli** [comment]
Have you tested? Does it help?

File: src/main/java/org/jclouds/vcloud/director/v1_5/compute/functions/ImageForVAppTemplate.java [hunk 1]
```diff
@@ -86,6 +90,17 @@ public Image apply(QueryResultVAppTemplateRecord from) {
       }
       builder.operatingSystem(os);
       builder.status(toPortableImageStatus.apply(from.getStatus()));
+
+      Reference reference = Reference.builder().name(from.getVdc()).href(URI.create(from.getVdc())).build();
+
+      if (reference != null) {
```
Comment (andreaturli) [inline src/main/java/org/jclouds/vcloud/director/v1_5/compute/functions/ImageForVAppTemplate.java]: can it be null?

File: src/main/java/org/jclouds/vcloud/director/v1_5/compute/functions/ImageForVAppTemplate.java [hunk 2]
```diff
@@ -86,6 +90,17 @@ public Image apply(QueryResultVAppTemplateRecord from) {
...
+         try {
+            builder.location(findLocationForResource.apply(reference));
+         } catch (NoSuchElementException e) {
+            logger.debug(e.getMessage());
```
Comment (andreaturli) [inline src/main/java/org/jclouds/vcloud/director/v1_5/compute/functions/ImageForVAppTemplate.java]: maybe it's a warning. add a message to warn users about the issue. something like, logger.warn("the image id doesn't have a location metadata attached and may lead to priority issues", e);

**Comment by bostko** [comment]
Is it possible to use a template from a different VDC with this PR?
I imagine that images from the VDC which is used should be used with priority but a jclouds should still be able to provision from other VDC.

**Comment by ivanayov** [comment]
@andreaturli I've tested and it works ok

@bostko a template from different VDC can be selected if there is missing location metadata for any image and there is no image attached with the current VDC. If all images have location and none of them has the current location, that means that the template is not shared with the current VDC and deployment should fail.
For allowing usage of template from another VDC when there is the same template from the current, IMO this should never be a case with priority. If we want to support this, we should have a discussion and may be point to another issue.

**Comment by andreaturli** [comment]
thanks @iyovcheva

**Pull request merged**